src/actions/save/index.js

```diff
@@ -51,22 +54,64 @@ const validateSaveConfig = (config: SaveActionConfig): void => {
 export const createSaveAction: SaveAction = (config: SaveActionConfig) => {
   validateSaveConfig(config)
 
+  const logger = getLogger()
+
   return {
-    type: "save",
-    save: (onError, paymentSourceDetails) => {
-      const { createVaultSetupToken } = config;
+    type: "SAVE",
+    save: (onError, paymentSourceDetails, lowScopedAccessToken) => {
+      const { createVaultSetupToken, onApprove } = config;
     
       if (!onError || !paymentSourceDetails) {
         return ZalgoPromise.reject(new ValidationError("Missing args to #save"))
       }
 
+      const handleError = (error) => {
+        logger.error('pp_sdk_actions_save_error', stringifyError(error))
+        onError(error.message)
+      }
+
       return ZalgoPromise.try(() => {
         createVaultSetupToken()
-        .then((/* emptySetupToken */) => {
+        .then(({ vaultSetupToken }) => {
+          // const vaultUrl = `${ getPayPalDomain() }/v3/vault/setup-tokens/${vaultSetupToken}/update`
+          const vaultUrl = `https://msmaster.qa.paypal.com:18582/v3/vault/setup-tokens/${vaultSetupToken}/update`
+          // call the vault api to update the setup token with our payment details
+          return request({
+            method: 'post',
+            url: vaultUrl,
+            headers: {
+              // Figure out where we can get authToken
+              'Authorization': `Basic ${lowScopedAccessToken}`,
```

Comment:
How is this type of thing handled already within the SDK?

Reply:
https://github.com/paypal/paypal-sdk-client/blob/main/src/api.js#L40